src/session.rs

```diff
@@ -32,12 +33,257 @@ pub(crate) enum Cmd {
         req: hyper::Request<hyper::Body>,
         rsp: oneshot::Sender<Result<hyper::Response<hyper::Body>, hyper::Error>>,
     },
-    WebDriver(Wcmd),
+    WebDriver(Box<dyn WebDriverCompatibleCommand + Send>),
+}
+
+impl WebDriverCompatibleCommand for Wcmd {
+    /// Helper for determining what URL endpoint to use for various requests.
+    ///
+    /// This mapping is essentially that of https://www.w3.org/TR/webdriver/#list-of-endpoints.
+    fn endpoint(
```

Comment:
I assume there weren't any changes in the logic in this method compared to what we had before?

Reply:
I had to change the bit at the top to go from `self.wdb` to `base_url` but other than that it's identical